java/java.editor.base/src/org/netbeans/modules/java/editor/base/semantic/SemanticHighlighterBase.java

```diff
@@ -657,6 +658,18 @@ public Void visitRequires(RequiresTree tree, Void p) {
             return super.visitRequires(tree, p);
         }
 
+        @Override
+        public Void visitBindingPattern(BindingPatternTree node, Void p) {
```

Comment:
I believe `when` is part of the `case` structure, not the binding pattern structure. Shouldn't the detection of the contextual keyword be in `visitCase`?

Reply:
@lahodaj : addressed

Reply:
Is there a need to keep the `when` handling here in `visitBindingPattern`? Also, it might make sense to also have a test for the deconstruction pattern with a guard.

Thanks!

Reply:
ok sure, will handle